geps/gep-2643/index.md

```diff
@@ -0,0 +1,439 @@
+      kinds:
+      - kind: TLSRoute
+    tls:
+      mode: Passthrough
+---
+apiVersion: gateway.networking.k8s.io/v1alpha3
+kind: TLSRoute
+metadata:
+  name: my-tls-route
+spec:
+  parentRefs:
+  - name: gateway-tlsroute
+  hostnames:
+  - foo.example.com
+  rules:
+  - backendRefs:
+    - name: tls-backend
+      port: 443
+```
+
+A typical [north/south](https://gateway-api.sigs.k8s.io/concepts/glossary/#northsouth-traffic)
+API request flow for a gateway implemented using a `TLSRoute` is:
+* A client makes a request to https://foo.example.com.
+* DNS resolves the name to a `Gateway` address.
+* The reverse proxy receives the request on a `Listener` and uses the
+[Server Name Indication](https://datatracker.ietf.org/doc/html/rfc6066#section-3)
+attribute to match an `TLSRoute`.
+* The reverse proxy passes through the request directly to one or more objects,
+i.e. `Service`, in the cluster based on `backendRefs` rules of the `TLSRoute`.
+
+### TLSRoute + TLS Termination
+In this workflow, the TLS traffic will be matched against the `SNI attribute` of 
+the request and terminated on the `Gateway`. 
+
+This workflow CAN be supported on Extended support level.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: gateway-tlsroute
+spec:
+  gatewayClassName: "my-class"
+  listeners:
+  - name: somelistener
+    port: 443
+    protocol: TLS
+    hostname: "*.example.com"
+    allowedRoutes:
+      namespaces:
+        from: Same
+      kinds:
+      - kind: TLSRoute
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - name: listener
+        kind: Secret
+---
+apiVersion: gateway.networking.k8s.io/v1alpha3
+kind: TLSRoute
+metadata:
+  name: my-rtmp-route
+spec:
+  parentRefs:
+  - name: gateway-tlsroute
+  hostnames:
+  - rtmp.example.com
+  rules:
+  - backendRefs:
+    - name: rtmp-backend
+      port: 12345
+```
+
+A typical [north/south](https://gateway-api.sigs.k8s.io/concepts/glossary/#northsouth-traffic)
+API request flow for a gateway implemented using both `TLSRoute` is:
+* A client makes a request to `rtmps://rtmp.example.com:443`.
+* DNS resolves the name to a `Gateway` address.
+* The reverse proxy receives the request on a `Listener` and uses the
+[Server Name Indication](https://datatracker.ietf.org/doc/html/rfc6066#section-3)
+attribute to match an `TLSRoute`.
+* The reverse proxy terminates the TLS negotiation on the `Gateway`.
+* The reverse proxy passes unencrypted request to one or more objects,
+i.e. `Service`, in the cluster based on `backendRefs` rules of the `TLSRoute`.
+
+### TLSRoute + Mixed Termination
+In this workflow, the TLS traffic will be matched against the `SNI attribute` of 
+the request, and based on the SNI attribute be directed to the backends on Passthrough
+mode or be terminated on the `Gateway` and passed unencrypted to the backends.
+
+This workflow CAN be supported on Implementation Specific support level and will
+be covered on a further GEP.
```

Comment:
So for now, if somebody tries to configure a mixed TLS termination on Gateway listeners on the same port, all such listeners will be marked as `Conflicted`. Standard conflict resolution doesn't work in this case as Gateway controller can't deterministically decide which listener was oldest for example (because listeners are defined inline within a single Gateway resource, they share the same creation timestamp).